### ISSUE TITLE
DisplaySettings: Crash when clicking not on a file

### ISSUE THREAD

**Issue by huttongrabiel** [SerenityOS/serenity#17935]
To reproduce:
1.  Open settings, open Display Settings
2.  Within the small file selector, click somewhere not on a file.

Bisected to ebb9c3a4304e9860386189bc6ac43a11f84f6ec1

**Comment by huttongrabiel** [comment]
@LucasChollet In case you know why this would cause issues. It crashes if you do above as well as if you try and select a file from Browse in DisplaySettings.

**Comment by LucasChollet** [comment]
Don't you have a PR that solve the issue?

**Comment by LucasChollet** [comment]
```diff
diff --git a/Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp b/Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp
index 6afe6f043f..757ae286dd 100644
--- a/Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp
+++ b/Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp
@@ -60,7 +60,6 @@ ErrorOr<void> BackgroundSettingsWidget::create_frame()
     m_wallpaper_view->set_model(GUI::FileSystemModel::create("/res/wallpapers"));
     m_wallpaper_view->set_model_column(GUI::FileSystemModel::Column::Name);
     m_wallpaper_view->on_selection_change = [this] {
-        String path = String::from_utf8_short_string(""sv);
         if (!m_wallpaper_view->selection().is_empty()) {
             auto index = m_wallpaper_view->selection().first();
             auto path_or_error = String::from_deprecated_string(static_cast<GUI::FileSystemModel*>(m_wallpaper_view->model())->full_path(index));
@@ -68,11 +67,10 @@ ErrorOr<void> BackgroundSettingsWidget::create_frame()
                 GUI::MessageBox::show_error(window(), "Unable to load wallpaper"sv);
                 return;
             }
-            path = path_or_error.release_value();
-        }
 
-        m_monitor_widget->set_wallpaper(path);
-        set_modified(true);
+            m_monitor_widget->set_wallpaper(path_or_error.release_value());
+            set_modified(true);
+        }
     };
 
     m_context_menu = GUI::Menu::construct();
```

AFAICT it solves the issue, but it would be nice to understand why it provokes a crash. I will try to do so later, but feel free to give it a shot. Also, can you confirm that the patch works correctly?

**Comment by huttongrabiel** [comment]
I'll take a look at this either later today or tomorrow and let you know.

**Comment by LucasChollet** [comment]
Fixed by #18015
